snyk/cli
# feat: build and test CLI fips binaries

#### What does this PR do?
* Build FIPS enabled CLI binaries for Linux and Windows (no user config necessary)
* Add FIPS related acceptance tests on Linux
* Add FIPS enabled Openssl to Linux build images
* Refactor pipeline jobs to use only a single build and acceptance test job for all OS/Archs
* Introduce e2e pattern for branch handling to run more steps of the release pipeline than a normal branch

#### Where should the reviewer start?
.circleci/config.yml

#### How should this be manually tested?
* Download the FIPS binaries from "Create Deployment Artifacts" job, they should panic when run on OSs that don't have FIPS enabled. Enable FIPS to see that the panic is gone.

#### Any background context you want to provide?
This uses the Microsoft go fork.

#### What are the relevant tickets?


#### Screenshots


#### Additional questions
